db_stress_tool/no_batched_ops_stress.cc

```diff
@@ -1870,7 +1870,7 @@ class NonBatchedOpsStressTest : public StressTest {
 
     // Write-prepared and Write-unprepared do not support Refresh() yet.
     if (!(FLAGS_use_txn && FLAGS_txn_write_policy != 0) &&
-        thread->rand.OneIn(2)) {
+        thread->rand.OneIn(2) && iter->IsAllowRefresh()) {
```

Comment:
Can we instead skip the validation when `iter->Refresh()` returns `NotSupported`? Then we don't need a new method in the `Iterator` class.

Reply:
@ajkr I changed the code, IMO the IsAllowRefresh() makes the code looks better, The condition is more readable (I removed the check for use_txm and txn_write_policy since both call NetItertator with allow_refresh_ = false.
trying to revert changes after shared->Get results errors.

Reply:
@ajkr I changed the code, IMO the IsAllowRefresh() makes the code looks better, The condition is more readable (I removed the check for use_txm and txn_write_policy since both call NetItertator with allow_refresh_ = false.
trying to revert changes after shared->Get results errors.